### ISSUE TITLE
[XPU] Segfault in WanVAE CausalConv3d during VAE encode/decode on Intel GPU

### ISSUE THREAD

**Issue by tinafengfun** [comfyanonymous/ComfyUI#14513]
# [XPU] Segfault in WanVAE CausalConv3d during VAE encode/decode on Intel GPU

## Summary

PyTorch Conv3d `_conv_forward` segfaults on Intel XPU when called by ComfyUI's `CausalConv3d.forward()` with `autopad="causal_zero"` weight slicing. This affects all WanVAE-based models (Wan 2.1/2.2, Qwen-Image) during VAE encode/decode on Intel Arc/Battlemage GPUs.

## Environment

- ComfyUI version: v0.22.0-124-gc87384f2 (commit `c87384f2b8f9e9a7f717e1a269b860698b6634bf`, 2026-06-02)
- PyTorch: 2.11.0+xpu
- Python: 3.13.3
- GPU: Intel(R) Graphics [0xe211] (PCI 8086:E211), 160 CUs, 24.4 GB VRAM
- Driver: xe kernel driver, Linux 6.14.0-1008-intel
- comfyui-frontend: 1.44.19

### Code path

```
VAEEncode → VAE.encode() → WanVAE.encode()
→ Encoder3d → CausalConv3d.forward(x) [x.shape[2]==1, fast path]
→ super().forward(x, autopad="causal_zero")    # vae.py:34
→ comfy.ops.Conv3d._conv_forward(weight slicing) # ops.py:596
→ torch.nn.functional.conv3d → SEGFAULT
```

When processing a single frame (T=1), `CausalConv3d.forward()` takes the fast path (`vae.py:34`):
```python
if cache_x is None and x.shape[2] == 1:
    return super().forward(x, autopad="causal_zero")
```

The `autopad="causal_zero"` triggers weight slicing in `comfy.ops.Conv3d._conv_forward` (`ops.py:596`), which slices the Conv3d weight tensor to simulate causal padding. The resulting `torch.nn.functional.conv3d` call segfaults on Intel XPU.

### Root cause

The combination of weight slicing + bf16 Conv3d on Intel XPU triggers a segfault in PyTorch's native Conv3d implementation. This is likely a bug in the Intel XPU SYCL backend for Conv3d with non-standard weight shapes or strides.

## Reproducer

```python
import torch, comfy.sd, comfy.utils

vae = comfy.sd.VAE(sd=comfy.utils.load_torch_file("qwen_image_vae.safetensors"))
model = vae.first_stage_model  # WanVAE
model.to('xpu').to(torch.bfloat16).eval()

x = torch.rand(1, 3, 1, 128, 128, dtype=torch.bfloat16, device='xpu') * 2.0 - 1.0
with torch.no_grad():
    out = model.encode(x)  # SEGFAULT here
```

Full reproducer: [repro1_vae_causal_conv.py](https://github.com/Comfy-Org/ComfyUI/blob/main/xpu-bug-investigation/repro1_vae_causal_conv.py) 

[repro1_vae_causal_conv.py](https://github.com/user-attachments/files/29035035/repro1_vae_causal_conv.py)

## Results

**3/3 deterministic segfaults** (exit code 139):

```
Run 1: Segmentation fault (exit 139)
Run 2: Segmentation fault (exit 139)
Run 3: Segmentation fault (exit 139)
```

Stack trace from faulthandler:
```
Fatal Python error: Segmentation fault

Current thread (most recent call first):
  File "repro1_vae_causal_conv.py", line 63 in main
    out = model.encode(x)
```

The segfault occurs inside native C code (torch Conv3d on XPU), so the Python traceback stops at the `.encode()` call.

## Existing workaround

ComfyUI already has a workaround at `comfy/sd.py:1143`:
```python
vae_cpu_offload = (self.device.type == 'xpu' and self.latent_dim >= 3)
```
This offloads VAE encode/decode to CPU for 3D-latent VAEs (WanVAE, Cosmos, etc.) on XPU. This works but significantly slows down the pipeline.

## Suggested fix

1. **Short-term**: The existing CPU offload workaround is sufficient for functionality. Consider logging a warning when it activates.
2. **Long-term**: Investigate the root cause in the PyTorch Intel XPU Conv3d backend. The issue may be related to non-contiguous weight tensors or unusual stride patterns from the causal weight slicing.
3. **PyTorch issue**: This should also be filed against `intel/intel-extension-for-pytorch` or the upstream PyTorch XPU backend.

## Impact

All workflows using WanVAE-based VAEs (Wan 2.1/2.2 video, Qwen-Image) on Intel XPU require CPU offload for VAE, adding ~5-15 seconds per encode/decode depending on resolution.

**Comment by rattus128** [comment]
Are you saying there is a workaround in place but you reverted it? The workaround you point is a hardcode.

**Comment by tinafengfun** [comment]
My workaround is to offload this model to cpu to avoid segment fault. BTW， this seems new issue found on Intel(R) oneAPI DPC++/C++ Compiler 2026.0.0 (2026.0.0.20260331).

**Comment by tinafengfun** [comment]
Will come up with a final fix later.

**Comment by tinafengfun** [comment]
> Are you saying there is a workaround in place but you reverted it? The workaround you point is a hardcode.

yes. I saw the work around from comfy but this may result in perf downgrade. I am working on pytorch xpu to resolve this issue completely. Once pytorch xpu fix is ready, I will update the ticket with the version for fix and validate to close the loop.  Thanks

**Comment by tinafengfun** [comment]
The issue was not reproduced on latest oneAPI setup. So close the issue.

**Comment by tinafengfun** [comment]
https://www.intel.com/content/www/us/en/developer/articles/release-notes/oneapi-toolkit/2026.html
